File: backend/app/schemas/matches.py
from pydantic import BaseModel
from typing import List
from datetime import datetime

class Match(BaseModel):
    id: int
    job_title: str
    company_name: str
    candidate_name: str
    compatibility_score: int  # 0-100
    skills_tags: List[str]
    location: str
    work_model: str
    salary_range: str
    match_reason: str
    created_at: datetime

class MatchResponse(BaseModel):
    id: int
    job_title: str
    company_name: str
    candidate_name: str
    compatibility_score: int
    skills_tags: List[str]
    location: str
    work_model: str
    salary_range: str
    match_reason: str
    created_at: datetime

File: frontend/src/lib/matches.ts
import { api } from '@/lib/api'

export interface Match {
  id: number
  job_title: string
  company_name: string
  candidate_name: string
  compatibility_score: number
  skills_tags: string[]
  location: string
  work_model: string
  salary_range: string
  match_reason: string
  created_at: string
}

export interface MatchResponse {
  matches: Match[]
  total_matches: number
  avg_compatibility: number
}

export const matchesService = {
  async getMatches(): Promise<Match[]> {
    const response = await api.get('/matches/')
    return response.data
  }
}
